spring-web/src/main/java/org/springframework/web/multipart/support/MultipartResolutionDelegate.java

```diff
@@ -156,10 +156,7 @@ private static boolean isPartArray(MethodParameter methodParam) {
 	private static Class<?> getCollectionParameterType(MethodParameter methodParam) {
 		Class<?> paramType = methodParam.getNestedParameterType();
 		if (Collection.class == paramType || List.class.isAssignableFrom(paramType)){
-			Class<?> valueType = ResolvableType.forMethodParameter(methodParam).asCollection().resolveGeneric();
-			if (valueType != null) {
-				return valueType;
-			}
+			return ResolvableType.forMethodParameter(methodParam).asCollection().resolveGeneric();
```

Comment:
Although the proposed changed is technically correct, please revert this for the sake of readability.